### ISSUE TITLE
[feature proposal] emoji support

### ISSUE THREAD

**Issue by pietroppeter** [nim-lang/nimforum#237]
Hi,
I recently published a small package to give nim emoji support, [nimoji](https://github.com/pietroppeter/nimoji). Yesterday I ran into [this](https://forum.nim-lang.org/t/6096#37898) post on the forum (cit. "[pls. imagine a "heart" icon here which to make/insert I'm too clueless]") and I realized that we could use nimoji to give the emoji support to the forum.

If you think it can be useful I can try working on a PR in the weekend.
Best,
Pietro

**Comment by Yardanico** [comment]
We already have "smile" support in the RST parser - see https://github.com/nim-lang/Nim/blob/devel/lib/packages/docutils/rst.nim#L75.
Maybe we should extend it instead?

(Nim forum uses that exact RST module for parsing posts from RST+Markdown into HTML)

**Comment by dom96** [comment]
Emoji keyboards are pretty prevalent nowadays, IMO there is not need to have parsing support for emojis when you can use a keyboard to insert them or just search for them online (which I do often).

What I would prefer in the long-term is to implement a nice editor UI that has a "Insert Emoji" button.

**Comment by pietroppeter** [comment]
Yes, I agree that in the end an emoji picker (and maybe also an emojiautocomplete) would be much better. Maybe one day I might try to have nimoji provide support for those type of features.

For reference, I checked how discourse does it and ran into this informative article: https://blog.discourse.org/2015/12/emoji-and-discourse/

**Comment by pietroppeter** [comment]
> We already have "smile" support in the RST parser - see https://github.com/nim-lang/Nim/blob/devel/lib/packages/docutils/rst.nim#L75.
> Maybe we should extend it instead?
> 
> (Nim forum uses that exact RST module for parsing posts from RST+Markdown into HTML)

I did not know that rst parser had already support of smilies, nice to know! :)

Regarding extending rstparse, thinking about it, it would mean a lot of work for a small benefit. Currently a user who wants to parse keywords such as `:wave:` into a unicode emoji can just use nimoji (on a file that would then be passed through doc generator or on a string that is then passed to rstparser). If we were to extend sdtlib's rstparser one would have to also change the parsing mechanism ([`parseSmiley`](https://github.com/nim-lang/Nim/blob/devel/lib/packages/docutils/rst.nim#L654) is currently iterating over all possible simleys and that would not be efficient for emojis) and also the rendering mechanism (smileys are rendered with gifs from what I see in the code).
